wdl-analysis/src/analyzer.rs

```diff
@@ -393,52 +423,79 @@ impl Analyzer {
         }
     }
 
-    /// Finds documents to analyze.
+    /// Adds documents to the analyzer.
     ///
     /// If a specified path is a directory, it is recursively searched for WDL
     /// documents.
     ///
-    /// Returns the URIs of the discovered documents.
-    pub async fn find_documents(paths: Vec<PathBuf>) -> Vec<Arc<Url>> {
-        RayonHandle::spawn(move || {
+    /// Returns an error if there was a problem discovering documents for the
+    /// specified paths.
+    pub async fn add_documents(&self, paths: Vec<PathBuf>) -> Result<()> {
+        // Start by searching for documents
+        let documents = RayonHandle::spawn(move || {
             let mut documents = Vec::new();
             for path in paths {
                 if path.is_file() {
-                    if let Some(uri) = path_to_uri(&path) {
-                        documents.push(uri.into());
-                    }
-
+                    documents.push(path_to_uri(&path).with_context(|| {
+                        format!(
+                            "failed to convert path `{path}` to a URI",
+                            path = path.display()
+                        )
+                    })?);
                     continue;
                 }
 
+                if !path.is_dir() {
+                    bail!("path `{path}` does not exist", path = path.display());
+                }
+
                 let pattern = format!("{path}/**/*.wdl", path = path.display());
                 let options = glob::MatchOptions {
                     case_sensitive: true,
                     require_literal_separator: false,
                     require_literal_leading_dot: true,
                 };
 
-                match glob::glob_with(&pattern, options) {
-                    Ok(paths) => {
-                        for uri in paths.filter_map(|p| match p {
-                            Ok(path) => path_to_uri(&path),
-                            Err(e) => {
-                                log::error!("error while searching for WDL documents: {e}");
-                                None
-                            }
-                        }) {
-                            documents.push(uri.into());
-                        }
-                    }
-                    Err(e) => {
-                        log::error!("error while searching for WDL documents: {e}");
+                let paths = glob::glob_with(&pattern, options).with_context(|| {
+                    format!("failed to glob directory `{path}`", path = path.display())
+                })?;
+                for file in paths {
+                    let file = file.with_context(|| {
+                        format!("failed to read directory `{path}`", path = path.display())
+                    })?;
+
+                    if !file.is_file() {
```

Comment:
Should we follow symlinks here? Or at least warn that we aren't.

Reply:
`PathBuf::is_file` does traverse symlinks.

Still, I'm going to rewrite this to use `walkdir` and not `glob` as the latter doesn't do symlink cycle detection.